### PR TITLE
Warn when not updating resources with init failures

### PR DESCRIPTION
**Pull request by hausdorff**

### CODE REVIEW

File: pkg/engine/plan.go [hunk 1]
```diff
@@ -223,6 +225,18 @@ func (res *planResult) Walk(ctx *Context, events deploy.Events, preview bool) (d
 				return
 			}
 
+			// Warn the user if they're not updating a resource whose initialization failed.
+			if step.Op() == deploy.OpSame && len(step.Old().InitErrors) > 0 {
+				prefix := colors.SpecImportant + "         [ERROR]" + colors.Reset
```
Comment (CyrusNajmabadi) [inline pkg/engine/plan.go]: it's a bit weird that we're mixing warning/[Error] here.  if this really is an error... then we should emit an error.  if it's not really an error, we should just do it as a warning (IMO) .

Comment (CyrusNajmabadi) [inline pkg/engine/plan.go]: can you add a screenshot to the description of what this looks like both during update and when update is complete.  i'd like to understand why we need to do things like add this whitespace here.

Comment (hausdorff) [reply pkg/engine/plan.go]: Sure:

screen shot 2018-07-20 at 1 29 00 pm

Comment (hausdorff) [reply pkg/engine/plan.go]: So what we want to communicate is:

1. That there were initialization errors previously.
1. What those initialization errors were.
1. You don't _have_ to update that resource to fix them but you definitely _should_.

I think I'm ok with pretty much any text that communicates these things effectively. If you have suggestions, let me know... the screenshot below gives a flavor of what it looks like now.

Comment (CyrusNajmabadi) [reply pkg/engine/plan.go]: I mean... we're already printing out that there's a warning here.  So i might just say that there's no need for the [Error] text.   If anything, we can say [Issue] if we want to be generic about it and we want to enuemrate the issues that happened.

Note: i find this text very clunky "failed with the following errors, but this plan does not update it".

Comment (CyrusNajmabadi) [reply pkg/engine/plan.go]: ok.  so my preference would be to not add the whitespace, but have that done at the presentation alyer.  i.e. if we have multiple lines, align them after the "error/warning/info<colon>" text.

Comment (CyrusNajmabadi) [reply pkg/engine/plan.go]:
```
warning: Previous initialization of this resource encountered problems.  Updating resource recommended to address this. 
         Issue #1: Yadda yadda 
         Issue #2: Yadda yadda
```

Comment (CyrusNajmabadi) [reply pkg/engine/plan.go]: This is minor though.  So i'm not going to gate the PR on this.  However, this sort of ting does impact the ability to change how the CLI works.  ON hte other hand, if the data is just a long string with newlines, the CLI can decide how to present it (including how ot indent multi-line constructs).

Comment (hausdorff) [reply pkg/engine/plan.go]: Ok, that sounds good, we'll use that wording.

Comment (hausdorff) [reply pkg/engine/plan.go]: Ok, we agreed offline to move that to a different review.

Comment (hausdorff) [reply pkg/engine/plan.go]: Looks like this now:

screen shot 2018-07-20 at 5 34 15 pm

**Comment by hausdorff** [comment]
Verbal approval from @CyrusNajmabadi

**Pull request merged**